version: '3'
services:
  sync:
    image: waisbrot/wait
    depends_on:
      - auth
      - call-logd
      - confd
      - postgres
      - rabbitmq
    environment:
      TARGETS: "postgres:5432 rabbitmq:5672 call-logd:9298 confd:9486 auth:9497"
      TIMEOUT: "${INTEGRATION_TEST_TIMEOUT}"

  auth:
    image: p0bailey/docker-flask
    ports:
      - "9497"
    volumes:
      - "./xivo-auth:/usr/local/share/xivo-auth"
      - "./ssl:/usr/local/share/ssl"
    command: "python /usr/local/share/xivo-auth/mock-xivo-auth.py 9497"

  call-logd:
    image: wazo-call-logd-test
    volumes:
      - "../../..:/usr/src/wazo-call-logd"
      - "./etc/wazo-call-logd:/etc/wazo-call-logd"
      - "./ssl:/usr/local/share/ssl"
      # - "${LOCAL_GIT_REPOS}/xivo-lib-python/xivo:/usr/local/lib/python3.5/site-packages/xivo"
      # - "${LOCAL_GIT_REPOS}/xivo-dao/xivo_dao:/usr/local/lib/python3.5/site-packages/xivo_dao"
      # - "${LOCAL_GIT_REPOS}/xivo-bus/xivo_bus:/usr/local/lib/python3.5/site-packages/xivo_bus"
    ports:
      - "9298"
    environment:
      XIVO_UUID: "08c56466-8f29-45c7-9856-92bf1ba89b92"

  confd:
    image: p0bailey/docker-flask
    ports:
      - "9486"
    volumes:
      - "./xivo-confd:/usr/local/share/xivo-confd"
      - "./ssl:/usr/local/share/ssl"
    command: "python /usr/local/share/xivo-confd/mock-xivo-confd.py 9486"

  postgres:
    image: wazopbx/postgres-test
    ports:
      - "5432"

  rabbitmq:
    image: rabbitmq
    ports:
      - "5672"
